lib/ansible/module_utils/urls.py

```diff
@@ -697,15 +697,11 @@ def http_request(self, req):
             # close the ssl connection
             #ssl_s.unwrap()
             s.close()
-        except (ssl.SSLError, socket.error):
-            e = get_exception()
-            # fail if we tried all of the certs but none worked
-            if 'connection refused' in str(e).lower():
-                raise ConnectionError('Failed to connect to %s:%s.' % (self.hostname, self.port))
-            else:
-                build_ssl_validation_error(self.hostname, self.port, paths_checked)
-        except CertificateError:
+        except (ssl.SSLError, CertificateError):
             build_ssl_validation_error(self.hostname, self.port, paths_checked)
+        except socket.error:
+            e = get_exception()
+            raise ConnectionError('Failed to connect to %s at port %s: %s' % (self.hostname, self.port, e))
```

Comment:
Imho, there is no need in such hack here, the only case I can see - it would fail if someone use non-english locale, but then this place would probably be last thing to worry about :-). Maybe we shall keep it without str() and to_native here?